lib/Jsrt/ChakraCommon.h

```diff
@@ -2978,4 +2978,343 @@ typedef unsigned char* ChakraBytePtr;
+    /// <param name="read">If the handle should be opened for reading.</param>
+    /// <param name="write">If the handle should be opened for writing.</param>
+    /// <returns>A HANDLE opened in read/write mode as specified.</returns>
+    typedef HANDLE (CALLBACK *JsTTDGetSrcCodeStreamCallback)(_In_z_ const wchar_t* uri, _In_z_ const wchar_t* bodyCtrId, _In_z_ const wchar_t* srcFileName, _In_ bool read, _In_ bool write);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     A callback for reading data from a handle.
+    /// </summary>
+    /// <param name="strm">The HANDLE to read the data from.</param>
+    /// <param name="buff">The buffer to place the data into.</param>
+    /// <param name="size">The max number of bytes that should be read.</param>
+    /// <param name="readCount">The actual number of bytes read and placed in the buffer.</param>
+    /// <returns>TRUE if the read was successful FALSE otherwise.</returns>
+    typedef BOOL(CALLBACK *JsTTDReadBytesFromStreamCallback)(_In_ HANDLE strm, _Out_writes_(size) BYTE* buff, _In_ DWORD size, _Out_ DWORD* readCount);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     A callback for writing data to a handle.
+    /// </summary>
+    /// <param name="strm">The HANDLE to write the data to.</param>
+    /// <param name="buff">The buffer to copy the data from.</param>
+    /// <param name="size">The max number of bytes that should be written.</param>
+    /// <param name="readCount">The actual number of bytes written to the HANDLE.</param>
+    /// <returns>TRUE if the write was successful FALSE otherwise.</returns>
+    typedef BOOL(CALLBACK *JsTTDWriteBytesToStreamCallback)(_In_ HANDLE strm, _In_reads_(size) BYTE* buff, _In_ DWORD size, _Out_ DWORD* writtenCount);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Flush and close the stream represented by the HANDLE as needed.
+    /// </summary>
+    /// <remarks><para>Exactly one of read or write will be set to true.</para></remarks>
+    /// <param name="read">If the handle was opened for reading.</param>
+    /// <param name="write">If the handle was opened for writing.</param>
+    typedef void (CALLBACK *JsTTDFlushAndCloseStreamCallback)(_In_ HANDLE strm, _In_ bool read, _In_ bool write);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Creates a new runtime in Record Mode.
+    /// </summary>
+    /// <param name="attributes">The attributes of the runtime to be created.</param>
+    /// <param name="infoUri">The uri where the recorded Time-Travel data should be stored.</param>
+    /// <param name="snapInterval">The interval to wait between snapshots (measured in millis).</param>
+    /// <param name="snapHistoryLength">The length of time to maintain before discadring (measured in number of snapshots).</param>
```

Comment:
discarding. This is kind of ring buffer right? if so a remarks explaining this would be helpful